tests/csapi/power_levels_test.go

```diff
@@ -40,3 +46,125 @@ func TestDemotingUsersViaUsersDefault(t *testing.T) {
 		},
 	})
 }
+
+func TestPowerLevels(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+
+	roomID := alice.CreateRoom(t, map[string]interface{}{})
+
+	// sytest: GET /rooms/:room_id/state/m.room.power_levels can fetch levels
+	t.Run("GET /rooms/:room_id/state/m.room.power_levels can fetch levels", func(t *testing.T) {
+		// Test if the old state still exists
+		res := alice.MustDoFunc(t, "GET", []string{"_matrix", "client", "v3", "rooms", roomID, "state", "m.room.power_levels"})
+
+		// note: before v10 we technically cannot assume that powerlevel integers are json numbers,
+		//  as they can be both strings and numbers.
+		// However, for this test, we control the test environment,
+		//  and we will assume the server is sane and give us powerlevels as numbers,
+		//  and if it doesn't, that's an offense worthy of a frown.
+
+		must.MatchResponse(t, res, match.HTTPResponse{
+			StatusCode: 200,
+			JSON: []match.JSON{
+				match.JSONKeyTypeEqual("ban", gjson.Number),
+				match.JSONKeyTypeEqual("kick", gjson.Number),
+				match.JSONKeyTypeEqual("redact", gjson.Number),
+				match.JSONKeyTypeEqual("state_default", gjson.Number),
+				match.JSONKeyTypeEqual("events_default", gjson.Number),
+				match.JSONKeyTypeEqual("users_default", gjson.Number),
+
+				match.JSONKeyTypeEqual("events", gjson.JSON),
+
+				match.JSONKeyTypeEqual("users", gjson.JSON),
+				match.JSONKeyTypeEqual("users."+client.GjsonEscape(alice.UserID), gjson.Number),
+
+				func(body []byte) error {
+					userDefault := int(gjson.GetBytes(body, "users_default").Num)
+					thisUser := int(gjson.GetBytes(body, "users."+client.GjsonEscape(alice.UserID)).Num)
+
+					if thisUser > userDefault {
+						return nil
+					} else {
+						return fmt.Errorf("expected room creator (%d) to have a higher-than-default powerlevel (which is %d)", thisUser, userDefault)
+					}
+				},
+			},
+		})
+	})
+
+	// sytest: PUT /rooms/:room_id/state/m.room.power_levels can set levels
+	t.Run("PUT /rooms/:room_id/state/m.room.power_levels can set levels", func(t *testing.T) {
+		// note: these need to be floats to allow a roundtrip comparison
+		PLContent := map[string]interface{}{
+			"invite": 100.0,
+			"users": map[string]interface{}{
+				alice.UserID: 100.0,
```

Comment:
I think this was missed from last time: I'd advise adding another user ID here (sytest used `@random-other-user:their.home` at power level 20) — that way, we can test that the server allows you to set *someone else's* power level (!)